### PR TITLE
Allow templating cluster-local AIA URIs

### PR DESCRIPTION
**Pull request by cipherboy**
This PR adds support for templating cluster-local and issuer-specific AIA URIs on leaf certificates. With the existing cluster-local CRL and OCSP responses, this lets correct AIA information be provided on requests serviced by PR Secondary clusters.

Notably, if the cluster-local mount path is not provided, and templating is enabled, issuance will fail.

Left to do:

 - [x] Tests
 - [x] Changelog entry
 - [x] Discuss if we should replace the warning with an error if templating fails on update.
 - [x] Is the behavior change around templating-enabled (with Issuer Ref) on root generation eliding AIA URIs OK? This occurs when `issuer_ref` is present in the template string but we don't yet know the issuer UUID. I don't think roots in general should really have AIA URIs, so I'm fine with that.

### CODE REVIEW

File: builtin/logical/pki/path_config_urls.go [hunk 1]
```diff
@@ -120,29 +131,65 @@ func (b *backend) pathWriteURL(ctx context.Context, req *logical.Request, data *
 		return nil, err
 	}
 
+	if enableTemplating, ok := data.GetOk("enable_templating"); ok {
+		entries.EnableTemplating = enableTemplating.(bool)
```
Comment (stevendpclark) [inline builtin/logical/pki/path_config_urls.go:135]: A thought, if we go from true to false on the EnableTemplating here, should we verify that the various urls do not have the replacement strings within them? I believe we will catch `{{cluster_path}}` but not `{{issuer_id}}` in the validation below.

Comment (stevendpclark) [reply builtin/logical/pki/path_config_urls.go:135]: Or possibly in general add that validation if `EnableTemplating` is false?

Comment (cipherboy) [reply builtin/logical/pki/path_config_urls.go:136]: Okie dokie, updated!

File: website/content/api-docs/secret/pki.mdx [hunk 2]
```diff
@@ -3101,6 +3132,76 @@ $ curl \
 }
 ```
 
+### Read Cluster Configuration
+
+This endpoint fetches the cluster-local configuration.
+
+Presently the only cluster-local config option is `path`, which sets the URL
+to this mount on a particular performance replication cluster. This is useful
+for AIA URL templating.
```
Comment (stevendpclark) [inline website/content/api-docs/secret/pki.mdx:3141]: Should we call the variable out explicitly, here and possibly below?

`This is useful for populating '{{cluster_path}}' within AIA URL templating`

**Review by stevendpclark — APPROVED** [review]

File: builtin/logical/pki/path_fetch_issuers.go [hunk 3]
```diff
@@ -732,6 +760,12 @@ func (b *backend) pathPatchIssuer(ctx context.Context, req *logical.Request, dat
 	if newName != oldName {
 		addWarningOnDereferencing(sc, oldName, response)
 	}
+	if issuer.AIAURIs != nil && issuer.AIAURIs.EnableTemplating && !b.useLegacyBundleCaStorage() {
```
[the remark this answers is not in the document]
Comment (cipherboy) [reply builtin/logical/pki/path_fetch_issuers.go:763]: Ah, right. :-)

File: website/content/api-docs/secret/pki.mdx [hunk 4]
```diff
@@ -2898,7 +2910,9 @@ parameter.
    suggested to use the per-issuer AIA information instead of the global
    AIA information. If any of the per-issuer AIA fields are set, the entire
    issuer's preferences will be used instead. Otherwise, these fields are used
-   as a fallback.
+   as a fallback.<br /><br />
+   This can be achieved by using _templated_ global AIA values, but setting
+   the cluster-local address in configuration.
```
[the remark this answers is not in the document]
Comment (cipherboy) [reply website/content/api-docs/secret/pki.mdx:2915]: FWIW, this was called out in the below docs on `enable_templating` but I've added a note here as well.

File: website/content/api-docs/secret/pki.mdx (hunk 2 above)
Comment (cipherboy) [reply website/content/api-docs/secret/pki.mdx]: Added, also added a notice that we may use this for other uses in the future.

**Review by stevendpclark — APPROVED** [review]

**Pull request merged**